### ISSUE TITLE
Transfer Learning Using Pretrained ConvNets tutorial should use sigmoid activation

### ISSUE THREAD

**Issue by jasonzhang2022** [tensorflow/tensorflow#28718]
## URL(s) with the issue:
https://www.tensorflow.org/alpha/tutorials/images/transfer_learning

## Description of issue (what needs changing):
classification head should use sigmoid activation.
### Clear description
The tutorial has this paragraph:

You don't need an activation function here because this prediction will be treated as a logit, or a raw prediciton value. Positive numbers predict class 1, negative numbers predict class 0.

I think we need sigmoid activation. 
Later we use loss="binary_crossentropy" in model.compile. binary_crossentropy by default has from_logits=False and expect a probability as is documented [here ](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/losses/BinaryCrossentropy) and [here](https://www.tensorflow.org/api_docs/python/tf/keras/losses/BinaryCrossentropy)

**Comment by MeghnaNatraj** [comment]
An activation adds uncertainty, causes increased model size and latency by a tiny bit -- so it's best if we avoid it. There is a workaround to set `from_logits=True` in the loss function as follows:

**Option 1 ([Source](https://www.tensorflow.org/tutorials/text/text_generation#attach_an_optimizer_and_a_loss_function))**
```
def loss(labels, logits):
  return tf.keras.losses.binary_crossentropy(labels, logits, from_logits=True)

model.compile(optimizer=tf.keras.optimizers.RMSprop(lr=base_learning_rate),
              loss=loss,
              metrics=['accuracy'])

```
**Option 2 (Optimized version of Option 1 using Lambda functions)**
```
model.compile(optimizer=tf.keras.optimizers.RMSprop(lr=base_learning_rate),
              loss=lambda labels, logits: tf.keras.losses.binary_crossentropy(labels, logits, from_logits=True),
              metrics=['accuracy'])
```

**Option 3: (Adheres to the [Keras documentation](https://keras.io/losses/#binary_crossentropy))**
```
model.compile(optimizer=tf.keras.optimizers.RMSprop(lr=base_learning_rate),
              loss=lambda y_true, y_pred: tf.keras.losses.binary_crossentropy(y_true, y_pred, from_logits=True),
              metrics=['accuracy'])
```

**Comment by rugg2** [comment]
**The incoherence is still present in the documentation**: still mentioning `activation=None` (logit) and `loss="binary_crossentropy"`.

This personally made me lose a couple of days of work, as one assumes TF's documentation is correct.

Happy with all the solutions suggested above. Another idea could be to use `binary_cross_entropy_with_logits` perhaps?

If somebody thinks the current tutorial is correct, it'd be great to have a thorough explanation. Otherwise, **can we have a hot fix?**

**Comment by MarkDaoust** [comment]
A fix is inflight,  Sorry for the confusion. 

The old version does work as intended, because setting loss to a string value triggers some keras magic that does the right thing behind the scenes.

But it's best to avoid magic.

**Comment by ymodak** [comment]
Closing this issue since the associated PR has been merged which fixes the issue. Thanks!
